src/librustc_mir/borrow_check/conflict_errors.rs

```diff
@@ -591,8 +600,8 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
     ) -> (String, String, String, String) {
         // Define a small closure that we can use to check if the type of a place
         // is a union.
-        let union_ty = |place: &Place<'tcx>| -> Option<Ty<'tcx>> {
-            let ty = place.ty(self.body, self.infcx.tcx).ty;
+        let union_ty = |place_base: &PlaceBase<'tcx>, place_projection: &Option<Box<Projection<'tcx>>>| -> Option<Ty<'tcx>> {
```

Comment:
I don't think these type annotations are necessary. `Place::ty_from` is nongeneric